From 39593e3d49d9259db17058b61a89193fb3cc6a40 Mon Sep 17 00:00:00 2001
From: Christophe Kerello <christophe.kerello@foss.st.com>
Date: Fri, 5 Apr 2024 11:42:54 +0200
Subject: [PATCH] dt-bindings: mmc: mmci: add address remapping property

Add new property st,syscfg-arcr to enable address remapping.
Remapping allows to access the 4 GB of the DDR.

Change-Id: Ibf87b60bfc6288e12e4ce7db0bd42c2981bda300
Signed-off-by: Christophe Kerello <christophe.kerello@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/371717
Reviewed-by: Yann GAUTIER <yann.gautier@foss.st.com>
Tested-by: Christophe KERELLO <christophe.kerello@st.com>
Domain-Review: Christophe KERELLO <christophe.kerello@st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Christophe KERELLO <christophe.kerello@st.com>
---
 Documentation/devicetree/bindings/mmc/arm,pl18x.yaml | 10 ++++++++++
 1 file changed, 10 insertions(+)

--- a/Documentation/devicetree/bindings/mmc/arm,pl18x.yaml
+++ b/Documentation/devicetree/bindings/mmc/arm,pl18x.yaml
@@ -167,6 +167,16 @@ properties:
     description:
       The GPIO matching the CKIN pin.
 
+  st,syscfg-arcr:
+    description: ST Micro-specific property, use to enable address remapping
+      to allow to access the 4 GB of the DDR.
+    $ref: /schemas/types.yaml#/definitions/phandle-array
+    items:
+      - items:
+          - description: phandle to syscfg
+          - description: register offset within syscfg
+          - description: register bitmask for address remapping bit
+
 dependencies:
   st,cmd-gpios: [ "st,use-ckin" ]
   st,ck-gpios: [ "st,use-ckin" ]
